### PR TITLE
BAVL-199: Recording the booking history for cancellation

### PR DESCRIPTION
**Pull request by timharrison-moj**
Tested locally.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/CancelVideoBookingService.kt [hunk 1]
```diff
@@ -12,13 +14,25 @@ import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.repository.VideoBooki
...
+  }
+
   @Transactional
   fun cancel(videoBookingId: Long, cancelledBy: String): VideoBooking {
     val booking = videoBookingRepository
       .findById(videoBookingId)
       .orElseThrow { EntityNotFoundException("Video booking with ID $videoBookingId not found.") }
 
-    return videoBookingRepository.saveAndFlush(booking.cancel(cancelledBy))
+    // TODO: Need to raise event to deal with cancelled or removing the remote appointments in A&A or NOMIS
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/CancelVideoBookingService.kt:31]: TODO not necessary.  Event is being raised.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/CancelVideoBookingServiceTest.kt [hunk 2]
```diff
@@ -31,6 +34,7 @@ class CancelVideoBookingServiceTest {
     booking.statusCode isEqualTo StatusCode.CANCELLED
 
     verify(videoBookingRepository).saveAndFlush(booking)
+    verify(bookingHistoryService).createBookingHistory(any(), any())
```
Comment (matt-crowson-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/CancelVideoBookingServiceTest.kt:37]: Why not `verify(bookingHistoryService).createBookingHistory(HistoryType.CANCEL, booking) `?

**Review by matt-crowson-moj — DISMISSED** [review]
LGTM.  A couple of very minor questions.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/CancelVideoBookingService.kt (hunk 1 above)
Comment (timharrison-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/CancelVideoBookingService.kt:31]: Is it? Can't see where?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/CancelVideoBookingServiceTest.kt (hunk 2 above)
Comment (timharrison-moj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/CancelVideoBookingServiceTest.kt:37]: For some reason the booking is not the object it expects... id is different. Assume something is creating a new version of it.

Comment (timharrison-moj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/CancelVideoBookingServiceTest.kt:37]: That was the case on Amend anyway, so might be an assumption here - I'll check.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/CancelVideoBookingService.kt (hunk 1 above)
Comment (timharrison-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/CancelVideoBookingService.kt:31]: I was expecting it to be in this service method, but its not.

Comment (timharrison-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/CancelVideoBookingService.kt:31]: BOOKING FACADE. Duh.

Comment (timharrison-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/CancelVideoBookingService.kt:31]: Removed TODO

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/CancelVideoBookingServiceTest.kt (hunk 2 above)
Comment (timharrison-moj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/CancelVideoBookingServiceTest.kt:37]: Not true here - so added the specific expected items and removed the any().

**Review by matt-crowson-moj — APPROVED** [review]
LGTM

**Pull request merged**